### PR TITLE
feat: adds theming homepage

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksadds-theming-homepage-hashicorp.vercel.app) 🔎
- [Asana task](https://app.asana.com/0/1204737346923690/1204737631862868) 🎟️
- [Figma File](https://www.figma.com/file/v0Hyqvjdf6QJAvRkdIBV24/Dark-Mode?type=design&node-id=1678-56758&t=ss97Qw0GIBxK8LXL-0) 🎨 

## 🗒️ What

Adds theming support to the homepage.

Screenshot 2023-06-20 at 4 27 18 PM

## 🛠️ How

Since we want to keep the homepage 'dark' in both dark and light themes, in dark mode we've done some overrides to switch tokens to the values that will retain the page styles. 

## 📸 Design Screenshots

Screenshot 2023-06-20 at 4 28 25 PM

## 🧪 Testing

- [Visit the homepage](https://dev-portal-git-ksadds-theming-homepage-hashicorp.vercel.app), toggle between themes
- Validate that the dark theme [matches designs](https://www.figma.com/file/v0Hyqvjdf6QJAvRkdIBV24/Dark-Mode?type=design&node-id=1678-56758&t=ss97Qw0GIBxK8LXL-0) and light theme design is retained

### CODE REVIEW

**Review by zchsh — APPROVED** [review]
✅ Code looks great to me!

🎨 Tested the home page in dark & light mode, looks great on that front too 🙌

**Review by kellyschummer — APPROVED** [review]
🥇 Beautiful!!

**Review by emilypersson1 — APPROVED** [review]
looks great, thank you!

File: src/views/homepage/index.tsx [hunk 1]
```diff
@@ -64,6 +64,5 @@ function HomePageView(): ReactElement {
  */
 HomePageView.contentType = 'tutorials'
 HomePageView.layout = BaseNewLayout
-HomePageView.theme = GlobalThemeOption.light
```
Comment (thiskevinwang) [inline src/views/homepage/index.tsx:67]: @kendallstrautman was this the last remaining instance where we set `.theme` on a page component?

**Pull request merged**